### PR TITLE
Feature deeplake

### LINKED ISSUE #17: Integrate with DeepLake
[DeepLake](https://github.com/activeloopai/deeplake) can function as a vector store. We want to add an integration to the worker to be able to write to DeepLake. 

DeepLake typically runs in your cloud, although there may be a way to run it locally with Docker, but not via a hosted, managed service. So this issue will also require bootstrapping DeepLake.

### PR DESCRIPTION
**Pull request by EteimZ**
## What

This PR is integrating deeplake  with vectorflow. The current integration only supports adding vectors to Deep Lake Storage. Subsequent PRs will integrate other [Storage options](https://docs.activeloop.ai/storage-and-credentials/storage-options).

## Why

It seeks to close #17 

## Usage

To work with this current implementation you will need an [activeloop(deeplake)](https://app.activeloop.ai/login/?utm_source=docs&utm_medium=gitbook&utm_campaign=button) account and you will need to create API tokens.

Specify your `vector_db_type` to be **DEEPLAKE**. Place your API key in the `X-VectorDB-Key` header.
Then your `index_name` should be of this format `hub://<active_loop_user_name>/<dataset_name>`. The dataset doesn't need to exist, deeplake will create it for you but you can always use existing dataset. 

Here's a sample request:

```code
curl -X POST -H 'Content-Type: multipart/form-data' -H "Authorization: my-internal-api-key" -H "X-EmbeddingAPI-Key: your-embedding-key" -H "X-VectorDB-Key: your-deeplake-api-key" -F 'EmbeddingsMetadata={"embeddings_type": "OPEN_AI", "chunk_size": 256, "chunk_overlap": 128}' -F 'SourceData=@./src/api/tests/fixtures/test_text.txt' -F 'VectorDBMetadata={"vector_db_type": "DEEPLAKE", "index_name": "hub://eteimz/vectorflow", "environment": "us-west1-gcp-free"}'  http://localhost:8000/embed
```

## Verification

To verify this PR works I created a file named `test_vectorflow.py` and added the following content to it:

```python
import deeplake
import os

token = os.environ['ACTIVELOOP_TOKEN']

ds = deeplake.load("hub://eteimz/my_vectorflow_dataset", token=token)
```

The environment variable contains my activeloop token. Running the script above yeilds this:

deeplake_integration_1

This basically says my specified dataset doesn't exist.

Then I use vectorflow to create that dataset:

deeplake_integration_2

I run the script a second time:

deeplake_integration_3

### CODE REVIEW

File: src/worker/vdb_upload_worker.py [hunk 1]
```diff
@@ -179,6 +182,45 @@ def write_embeddings_to_weaviate(text_embeddings_list, vector_db_metadata,  batc
     logging.info(f"Successfully uploaded {len(text_embeddings_list)} vectors to Weaviate")
     return len(text_embeddings_list)
 
+def write_embeddings_to_deeplake(text_embeddings_list, vector_db_metadata, batch_id, job_id):
+    ids = []
+    source_texts = []
+    embeddings = []
+
+    for i, (source_text, embedding) in enumerate(text_embeddings_list):
```
Comment (dgarnitz) [inline src/worker/vdb_upload_worker.py:190]: Please break this into its own method like how it is for the other vector DBs. That way we can unit test it down the road

File: docker-compose.yml [hunk 2]
```diff
@@ -89,6 +89,8 @@ services:
       context: ./src
       dockerfile: worker/Dockerfile.vdb-upload-worker
     container_name: vectorflow_vdb_upload_worker
+    volumes:
+      - ./src/:/app
```
Comment (dgarnitz) [inline docker-compose.yml:93]: @EteimZ  thanks for figuring this out!! Does this still enable running in detached mode or will we always have to have the terminal window that runs `docker-compose up` open for it to work/?

Comment (dgarnitz) [reply docker-compose.yml:93]: just want to understand a bit more if it should be in the default workflow, and if it is, we should add it to all the containers where its relevant

**Review by dgarnitz — CHANGES_REQUESTED** [review]
Excellent work. Truly impressed with how fast you did this. Just a few small changes it it will be ready

**Comment by dgarnitz** [comment]
@EteimZ is there a command or api call you can run in a separate script to verify that the data made it into DeepLake? If so, can you screenshot that and add it as verification evidence? Thanks

File: docker-compose.yml (hunk 2 above)
Comment (EteimZ) [reply docker-compose.yml:93]: This method works even in detached mode. The only draw back is I haven't figured out how to properly set it up with my IDE. So currently I am not getting autocomplete and intellisense while developing with it. I'm using this [resource](https://code.visualstudio.com/docs/devcontainers/tutorial) to try to get autocomplete and intellisense set up on vscode.

**Review by dgarnitz — APPROVED** [review]
Great work!

**Pull request merged**